### ISSUE TITLE
Switch credential after first DB init not working

### ISSUE THREAD

**Issue by databird** [SergeyTsalkov/meekrodb#106]
Hello,

We try, in our code, to switch to another database with different credential, without success :

```
// First init
DB::$user = 'user1';
DB::$password = 'pwd1';
DB::$dbName = 'db1';
... do stuff with db1

// Then try to connect to another database, with other credential
DB::disconnect();
DB::$user = 'user2';
DB::$password = 'pwd2';
DB::$dbName = 'db2'; //or DB::useDB('db2');
```

But credential are not updated : the connection to 'db2' is still using user1/pwd1 !

To make it working, we needed to manipulate credential through function :
```
DB::getMDB()->user = 'user2';
DB::getMDB()->password = 'pwd2';
```

Could it be possible, after doing a "DB::disconnect", when updating credential, that any new connection is updated with new credentials ?

**Comment by danielscsc** [comment]
I couldn't reproduce this on `v3.0.0`.  What version of Meekro are you using?

**Comment by databird** [comment]
Ahah good point ! We are indeed using version 2.5.1... :-)

We will try to upgrade, but passing to v3 break a lot of code that are working on 2.5 :-(... (do you have a guide about this migration somewhere?)

**Comment by SergeyTsalkov** [comment]
It looks like your workaround is pretty good-- I can't really put out a formal fix because it's already fixed in the latest release.

There is no migration guide for v3 because I thought it would be a 100% drop-in replacement. If you gave examples of stuff that worked before but doesn't work now, that would be pretty interesting to see. Maybe I can improve backwards compatibility in v3 somehow?

**Comment by databird** [comment]
We have this kind of code:
`$where->add( DB::parse('(PL.date_end IS NULL OR PL.date_end = "0000-00-00" OR PL.date_end >= %t)', $start_invoice) );`

That cause error:
```
array_values(): Argument #1 ($array) must be of type array, string given

/vendor/sergeytsalkov/meekrodb/db.class.php : 1377

#0 /vendor/sergeytsalkov/meekrodb/db.class.php(1377): array_values('2024-05-01 00:0...')
#1 /vendor/sergeytsalkov/meekrodb/db.class.php(1373): MeekroDBParsedQuery->add('?', '2024-05-01 00:0...')
#2 /vendor/sergeytsalkov/meekrodb/db.class.php(916): MeekroDBParsedQuery->add(Object(MeekroDBParsedQuery))
#3 /vendor/sergeytsalkov/meekrodb/db.class.php(112): MeekroDB->parse('(PL.date_end...', Object(DateTime))
#4 /invoices.php(31): DB::__callStatic('parse', Array)
```

And it was working on previous version ;-)

**Comment by SergeyTsalkov** [comment]
Oh, no wonder you're having problems.. you're relying on parse(), which is undocumented internal functionality. Its behavior has changed in the v3 release.

In your example, it's also not necessary. This will work:

```php
$where->add('PL.date_end IS NULL OR PL.date_end = "0000-00-00" OR PL.date_end >= %t', $start_invoice);
```

**Comment by databird** [comment]
I was using DB::parse because it's documented here : https://meekro.com/docs/misc-methods.html :-D
```
DB::parse()
Parse a query with args just like DB::query() would, but return it instead of executing
```

Thanks for you help about modifying `$where->add`, I didn't catch this behaviour 👍 

But sometime we also use DB::parse for preparing a part of a request:
```
$join = DB::parse(' LEFT JOIN placements P ON (P.`uid_contact` = C.uid AND P.status <> %s)', $status);
$join .= DB::parse(' LEFT JOIN placements_completed PC ON (PC.`uid_placement` = P.`uid` AND PC.date_end >= %t AND PC.date_start <= %t AND PC.type = %s AND (PC.uid_invoice IS NULL OR PC.uid_invoice = ""))', $start_invoice, $end_invoice, $type);

// and then, use $join in a request with literal (%l) placeholder:
$contacts = DB::query('SELECT C.*, PC.* FROM contacts C %l WHERE ...', $join, $where_arg1, $where_arg2, $where_arg3);
```
The goal is to have better visibility on code to not having 20+ arguments on DB::query

Any ideas how can we change our DB::parse call ?

**Comment by SergeyTsalkov** [comment]
Shit, you're right, I forgot that it was actually documented, and had no idea that people were relying on it to do anything serious.

I think I can actually fix this so your old code works again. I can probably do that tomorrow.

**Comment by SergeyTsalkov** [comment]
I published a new v3.0.1 release addressing this. What you're trying to do with parse() should work now. Keep in mind that you cannot use .= to concatenate the outputs of parse(), because they are not strings. However, you can include them in a query with %l or %?

Example:

```php
$part = DB::parse('username=%s', 'Joe');
$results = DB::query("SELECT * FROM accounts WHERE %?", $part);
```
